cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ModelUtils.java

```diff
@@ -26,6 +29,25 @@ public static void init(KafkaCruiseControlConfig config) {
     _useLinearRegressionModel = config.getBoolean(KafkaCruiseControlConfig.USE_LINEAR_REGRESSION_MODEL_CONFIG);
   }
 
+  /**
+   * Get a single snapshot value that is representative for the CPU leadership utilization. The current algorithm uses
+   * the mean of the recent CPU leadership load.
+   *
+   * @param leaderLoad Load of the leader replica.
+   * @return A single representative utilization value on CPU leadership load.
+   */
+  public static double expectedLeadershipCpuUtilization(Load leaderLoad) {
+    List<Snapshot> snapshotsByTime = leaderLoad.snapshotsByTime();
+    double accumulatedUtilization = 0.0;
+
+    for (Snapshot snapshot : snapshotsByTime) {
```

Comment:
This is going to slow down the proposal computation dramatically because it is going to be called for each leader movement in `CapGoal.isProposalAcceptable()`.

Reply:
Fixed. 
To optimize CC performance, we avoid calculation of the expected leadership CPU utilization, and assume that if `(proposal.balancingAction().equals(BalancingAction.LEADERSHIP_MOVEMENT) && resource().equals(Resource.CPU))`, then the expected leadership CPU utilization would be the full CPU utilization of the leader.